### ISSUE TITLE
Perform KDIndex searches using PointId and/or PointRef

### ISSUE THREAD

**Issue by chambbj** [PDAL/PDAL#1372]
Although query points for the `KDIndex` are not required to come from the indexed cloud, they often do, and developers using it this way could simplify their code quite a bit if they were allowed to provide a `PointId` from the indexed cloud. Something like...

```
KD3Index index(view);
index.build();
for (PointId i = 0; i < view.size(), ++i)
{
    std::vector<PointId> pts = index.radiusByPointId(i, 1.0);
}
```

Whether or not the query point came from the indexed cloud, another handy shortcut would be to allow developers to provide a `PointRef` instead of the x, y, z coordinates. Something like...

```
KD3Index index(view);
index.build();
for (PointId i = 0; i < view.size(), ++i)
{
    PointRef query(view, i);
    std::vector<PointId> pts = index.radiusByPointRef(query, 1.0);
}
```

**Comment by abellgithub** [comment]
Seems like this could be handled with overloads.  Also not sure why you'd use a PointRef if you just have to construct it from the index.  The KDIndex is already tied to a PointView.

**Comment by chambbj** [comment]
Overloads would be fine. `PointRef` may just be slightly simpler in the case that you were not drawing from the indexed cloud (though this isn't what is shown in my example).

**Comment by chambbj** [comment]
Fixed in #1377 via 035ebef17e43c4297f534ab40c941ed912b369d3
